### ISSUE TITLE
Use without global variables

### ISSUE THREAD

**Issue by cobraz** [ferdikoomen/openapi-typescript-codegen#428]
Hello,

I realize after using this library that using global variables is suboptimal for what I'm using it for.

In my case, I have a Node.js application. I need to run multiple processes with different auth (API tokens) for each, and having a global scope would limit my performance quite a bit (i have to run all in sequence to make sure that the right token is used for the request).

For quite some time, I used another library to generate code. However, that had classes for each service (like yours), but not with singletons. You had to construct each class and pass auth. E.g. `new PostService('token')`.

Code that handled more than one service would have to construct a few classes. Also sub-optimal. IMO, for the best possible API, I'd like to have one class that reference others.

Example:

```typescript
class MyApi {

	private auth;

	constructor(... some auth thing ...) {}

	postService() {
		return new PostService(auth);
	}
}
```

In summary, I'd love to see a way where this library could a) generate classes without singletons and b) have a class that exports constructed services.

I realize looking at your code that this isn't much work to do. Would you accept a pull request with that? I could also fork, but I would much rather help out :)

**Comment by cobraz** [comment]
Actually, you could probably introduce some configuration, not just auth, as such:

```
class MyApi {

	private config;

	constructor(... some config ...) {}

	postService() {
		return new postService(auth);
	}
}
```

**Comment by ferdikoomen** [comment]
@cobraz If you generate two OpenAPI clients and use a default (webpack, parcel, rollup) bundler then each client can have its own module scope. I will attach an example of the setup in the ticket. This is a much neater setup then exposing classes for this, otherwise you will need to pass around this MyApi class everywhere.

**Comment by vitalybaev** [comment]
Another good reason not to use global variables is SSR - using global variables to update token can cause issues when some user could occasionally become logged in as another user.

**Comment by ferdikoomen** [comment]
@vitalybaev @cobraz There is a new version (0.16.0) that allows to create client instances to each client can have its own bearer token. You can specify the `--name` argument while generating the client. More info:
https://github.com/ferdikoomen/openapi-typescript-codegen#generate-client-instance-with---name-option

**Comment by mb21** [comment]
For us, the `--name` option was not flexible enough, so we let the user pass in a whole template for the services: https://github.com/ferdikoomen/openapi-typescript-codegen/pull/1268

**Comment by silasdavis** [comment]
> otherwise you will need to pass around this MyApi class everywhere.

I'd much rather pass around state expicitly than have an implicit global state. How about a constructor that takes an optional `config: OpenAPIConfig = OpenAPI`, best of both worlds, no?
